### PR TITLE
Added HTTPMethod Enum to Class Rule

### PR DESCRIPTION
**Pull request by dvaseekara**

### CODE REVIEW

File: default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml [hunk 1]
```diff
@@ -49,3 +49,21 @@
...
+
+- ruleID: spring-framework-5.x-to-6.0-web-applications-00002
+  category: mandatory
+  effort: 1
+  labels:
+  - konveyor.io/source=spring5
+  - konveyor.io/target=spring6+
+  when:
+    java.referenced:
+      pattern: org.springframework.http.HttpMethod
```
Comment (jmle) [inline default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml:61]: Here we have two options:
a. Leave it as it is, and change the category to "potential". We would get many false positives.
b. Switch it to something like:
```yaml
    java.referenced:
      pattern: EnumSet<HttpMethod>
      location: TYPE
```
This wouldn't catch usages of switches, but we would get less false positives. What do you think @rromannissen ?

**Review by jmle — CHANGES_REQUESTED** [review]
@dvaseekara thanks for your first rule! I have a few suggestions.

File: default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml [hunk 2]
```diff
@@ -49,3 +49,21 @@
...
+      location: IMPORT
+  description: Refactor HttpMethod from enum to class
```
[the remark this answers is not in the document]
Comment (dvaseekara) [reply default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml]: Good policy, upated PR

File: default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml [hunk 3]
```diff
@@ -49,3 +49,21 @@
...
+  message: "HTTPMethod has been updated from an ENUM to a CLASS"
```
[the remark this answers is not in the document]
Comment (dvaseekara) [reply default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml]: Updated description to be more wordy

File: default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml (hunk 1 above)
Comment (dvaseekara) [reply default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml:91]: From what I'm thinking, this is a pretty minor change when it comes to individual instances of using the ENUM.  So calling it out once for the whole CLASS as a "1" effort seems more appropriate than calling out each instance.

File: default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml [hunk 4]
```diff
@@ -49,3 +49,21 @@
     url: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-6.0-Release-Notes#web-applications
   - title: 'Specific Spring Framwork change'
     url: https://github.com/spring-projects/spring-framework/issues/28552
+
+- ruleID: spring-framework-5.x-to-6.0-web-applications-00002
+  category: mandatory
```
[the remark this answers is not in the document]
Comment (dvaseekara) [reply default/generated/spring-framework/spring-framework-5.x-to-6.0-web-applications.yaml:55]: Updated to potential

**Comment by rromannissen** [comment]
Fixes https://github.com/konveyor/rulesets/issues/153

**Comment by jmle** [comment]
@dvaseekara please sign the commits of the PR. You can squash them previously into a single one if you want.

**Pull request closed without merging**